osu.Game/Overlays/Settings/Sections/Maintenance/DirectorySelectScreen.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using System.IO;
+using osu.Framework.Allocation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.UserInterfaceV2;
+using osu.Game.Screens;
+using osuTK;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Graphics.UserInterface;
+using osu.Framework.Screens;
+
+namespace osu.Game.Overlays.Settings.Sections.Maintenance
+{
+    public abstract class DirectorySelectScreen : OsuScreen
+    {
+        private TriangleButton selectionButton;
+
+        private DirectorySelector directorySelector;
+
+        protected override OverlayActivation InitialOverlayActivationMode => OverlayActivation.Disabled;
+
+        protected abstract OsuSpriteText CreateHeader();
```

Comment:
Could also get away with a `protected string Header { get; }` but I'm not super fussed.

Reply:
Rationale for going with `CreateHeader()` was that some long header texts may require a smaller font size to not overflow from the dialog, but in practice it doesn't so yeah reverting to using a string property should be fine.

Reply:
Fair enough. I'd leave as is.

Reply:
oops i changed this.

if we get to the point where things won't fit then it should be changed to flowing text, rather than the size adjusted locally.

Reply:
aha, turns out this will be required far sooner than anticipated

![image](https://user-images.githubusercontent.com/20256717/118482182-ab885680-b714-11eb-8c73-eb49ed96e3ec.png)